driver-core/src/main/com/mongodb/connection/ConnectionDescription.java

```diff
@@ -120,6 +122,23 @@ public ConnectionDescription(final ConnectionId connectionId, final ServerVersio
         this.compressors = notNull("compressors", Collections.unmodifiableList(new ArrayList<String>(compressors)));
     }
 
+    /**
+     * Construct an instance.
+     *
+     * @param connectionId    the connection id
+     * @param maxWireVersion  the max wire version
+     * @param serverType      the server type
+     * @param maxBatchCount   the max batch count
+     * @param maxDocumentSize the max document size in bytes
+     * @param maxMessageSize  the max message size in bytes
+     * @param compressors     the available compressors on the connection
+     * @since 3.10
+     */
+    public ConnectionDescription(final ConnectionId connectionId, final int maxWireVersion,
```

Comment:
The driver doesn't use this new method yet as it still has to support the serverVersion property, but in 4.0 it will.  This gives any applications that happen to be creating ConnectionDescription instances a non-deprecated method to call (though there is no real use for an application to do that except for some sort of test scaffolding).